src/ngResource/resource.js

```diff
@@ -701,12 +701,9 @@ angular.module('ngResource', ['ng']).
               response.resource = value;
 
               return response;
-            }, function(response) {
-              (error || noop)(response);
-              return $q.reject(response);
             });
 
-            promise['finally'](function() {
+            promise = promise['finally'](function() {
```

Comment:
I meant (assuming that the finally callback does not return a promise and does not throw any errors), does it make any difference to have:

```js
promise.finally(...)
promise.then(...)
```
vs
```js
promise = promise.finally(...)
promise.then(...)
```

But thinking about it again, the former has the issue that is creates a new promise that will remain unhandled if `$http` rejects (as you explained in another comment).